databrowser/src/domain/datasets/ui/tableView/filter/tableFilterStore.ts

```diff
@@ -111,18 +115,28 @@ export const useTableFilterStore = defineStore('tableFilterStore', () => {
     value: FilterValue,
     applyFilter = true
   ) => {
-    const updatedFilters = tableFilters.value.map((filter, i) =>
+    tableFilters.value = tableFilters.value.map((filter, i) =>
       i === index ? { ...filter, operator, value } : filter
     );
 
-    tableFilters.value = updatedFilters;
-
     // Apply filters to URL such that they take effect
     if (applyFilter) {
       datasetFilters.value = tableFilters.value;
     }
   };
 
+  const updateFilterValueByAllSelectedTable = (
+      tableFiltersSelected: Filter[], // Array of strings
+      applyFilter = true
+  ) => {
+      tableFilters.value = tableFiltersSelected
```

Comment:
Formatting issue (intendation and missing semicolon at the end). Please check your tools, they should automatically resolve this kind of issues.